src/main/java/com/spikes2212/util/smartmotorcontrollers/SparkFlexWrapper.java

```diff
@@ -0,0 +1,114 @@
+    public SparkFlexWrapper(int deviceID, MotorType type, FeedForwardController feedForwardController) {
+        super(deviceID, type);
+        this.feedForwardController = feedForwardController;
+        sparkConfig = new SparkFlexConfig();
+        closedLoopConfig = new ClosedLoopConfig();
+    }
+
+    public SparkFlexWrapper(int deviceID, MotorType type) {
+        this(deviceID, type, new FeedForwardController(FeedForwardSettings.EMPTY_FFSETTINGS,
+                FeedForwardController.DEFAULT_PERIOD));
+    }
+
+    public SparkFlexConfig getSparkConfiguration() {
+        return sparkConfig;
+    }
+
+    public void applyConfiguration(SparkFlexConfig newConfig) {
+        sparkConfig.apply(newConfig);
+        configure(sparkConfig, ResetMode.kNoResetSafeParameters, PersistMode.kNoPersistParameters);
+    }
+
+    public ClosedLoopConfig getClosedLoopConfiguration() {
+        return closedLoopConfig;
+    }
+
+    public void applyClosedLoopConfig(ClosedLoopConfig newConfig) {
+        closedLoopConfig.apply(newConfig);
+        configure(sparkConfig.apply(closedLoopConfig), ResetMode.kNoResetSafeParameters,
+                PersistMode.kNoPersistParameters);
+    }
+
+    public void restoreFactoryDefaults() {
+        sparkConfig = new SparkFlexConfig();
+        closedLoopConfig = new ClosedLoopConfig();
+        configure(sparkConfig, ResetMode.kResetSafeParameters, PersistMode.kPersistParameters);
+    }
+
+    public void setInverted(boolean inverted) {
+        int leaderID = configAccessor.getFollowerModeLeaderId();
+        if (leaderID != 0) {
+            sparkConfig.follow(leaderID, inverted);
+        } else sparkConfig.inverted(inverted);
+        configure(sparkConfig, ResetMode.kNoResetSafeParameters, PersistMode.kNoPersistParameters);
+    }
+
+    public void follow(SparkBase master) {
+        configure(sparkConfig.follow(master, configAccessor.getInverted()), ResetMode.kNoResetSafeParameters,
+                PersistMode.kNoPersistParameters);
+    }
+
+    public void unfollow() {
+        sparkConfig.disableFollowerMode();
+        configure(sparkConfig, ResetMode.kNoResetSafeParameters, PersistMode.kNoPersistParameters);
+    }
+
+    public void setIdleMode(SparkBaseConfig.IdleMode idleMode) {
+        sparkConfig.idleMode(idleMode);
+        configure(sparkConfig, ResetMode.kNoResetSafeParameters, PersistMode.kNoPersistParameters);
+    }
+
+    @Override
+    public void configurePID(PIDSettings pidSettings) {
+        closedLoopConfig.pid(pidSettings.getkP(), pidSettings.getkI(), pidSettings.getkD());
+        configure(sparkConfig.apply(closedLoopConfig), ResetMode.kNoResetSafeParameters,
+                PersistMode.kNoPersistParameters);
+    }
+
+    @Override
+    public void configureFF(FeedForwardSettings feedForwardSettings) {
+        feedForwardController.setGains(feedForwardSettings);
+    }
+
+    @Override
+    public void configureTrapezoid(TrapezoidProfileSettings trapezoidProfileSettings) {
```

Comment:
i saw in the API that there is something called position control mode, is it relevant?